components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/ResponseTypeHandler.java

```diff
@@ -34,7 +35,7 @@ public boolean validateScope(OAuthAuthzReqMessageContext oauthAuthzMsgCtx)
             throws IdentityOAuth2Exception;
 
     public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx)
-            throws IdentityOAuth2Exception;
+            throws IdentityOAuth2Exception, InvalidOAuthClientException;
```

Comment:
This is an **API change**. 
Why do we need to throw InvalidOAuthClientException? Can't we wrap it in an IdentityOAuth2Exception?